### PR TITLE
Add Organization schema

### PR DESCRIPTION
**Pull request by cmgiven**
connects #91 

Sites are grouped into organizations. A user might have permissions at the per-site level, or the organization level.

Doing this now because the initial type of report is at the organization level, and it seemed like this would be harder to retrofit after the fact.

Notably there's now some inheritance going on in the Permissions models. I'm expecting to do something similar with Report. I was looking for how EF handles polymorphic relationships, and this seems to be the preferred approach (yes, not technically polymorphic). Docs here: https://docs.microsoft.com/en-us/ef/core/modeling/relational/inheritance

### CODE REVIEW

File: test/HedwigTests/Integration/GraphQLQueries/UserQueryTests.cs [hunk 1]
```diff
@@ -9,17 +9,18 @@
 
 namespace HedwigTests.Integration.GraphQLQueries
 {
-    [Collection("SqlServer")]
-    public class UserQueryTests
-    {
+	[Collection("SqlServer")]
```
Comment (jlhogan) [inline test/HedwigTests/Integration/GraphQLQueries/UserQueryTests.cs]: this collection annotation should have been removed in my #79 PR; can you delete it for me ?

**Comment by cmgiven** [comment]
It's definitely the case that some users have permissions only to some sites. I.e. I'm responsible for maintaining the roster for a particular site, but I can't update the other sites. I did debate with myself whether that eventuality was too far in the future to worry about here, but landed on the side of it being useful to play that out.

**Comment by cmgiven** [comment]
@speedcm Can you check SonarCloud for me? The code coverage doesn't appear to be working. The new lines in SiteRepository are definitely covered. But also it appears to be looking at code coverage of migrations and seed data, which should be ignored.

File: test/HedwigTests/Integration/GraphQLQueries/UserQueryTests.cs (hunk 1 above)
Comment (cmgiven) [reply test/HedwigTests/Integration/GraphQLQueries/UserQueryTests.cs]: Can do.

**Comment by speedcm** [comment]
To exclude directory or directories, add the extraProperties element (see sample below). 
 
 - task: SonarCloudPrepare@1
    inputs:
      SonarCloud: "sonar-cloud-ctoec"
      organization: "ctoec"
      scannerMode: "MSBuild"
      projectKey: "ecis-experimental"
      extraProperties: |
         sonar.exclusions="**/Migrations/*"

https://community.sonarsource.com/t/analysis-running-in-azure-pipeline-ignoring-file-exclusions/7416

**Comment by cmgiven** [comment]
And how about this one? https://sonarcloud.io/component_measures?id=ecis-experimental&metric=new_lines_to_cover&pullRequest=92&selected=ecis-experimental%3ARepositories%2FSiteRepository.cs&view=list

**Comment by speedcm** [comment]
In order to update SonarCloud for Code Coverage, we need to specify the reports path.  Can you add the following line under extraProperties. 

sonar.cs.vstest.reportsPaths="**\TestResults\*.trx"

Please adjust the path accordingly.

**Comment by cmgiven** [comment]
Can you clarify? Do you want exactly that line in the SonarCloud config? Or does this require more thought? If the latter, I'd suggest we create a new ticket for that, and disable code coverage checks until we have that figured out. @speedcm

**Comment by speedcm** [comment]
I also agree the quality check should be disabled.  I will create the ticket.  Looks like you will have to disable the quality gate in SonarCloud since you are the administrator.

**Pull request merged**